cmd/tempo-cli/cmd-query-blocks-summary.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+
+	"context"
+	"fmt"
+
+
+
+
+	"github.com/grafana/tempo/pkg/model/trace"
+
+	//"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/pkg/util"
+	// "github.com/grafana/tempo/tempodb"
+	// "github.com/grafana/tempo/tempodb/backend"
+	// "github.com/grafana/tempo/tempodb/encoding"
+	// "github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type queryBlocksSummaryCmd struct {
+	backendOptions
+
+	TraceID  string `arg:"" help:"trace ID to retrieve"`
+	TenantID string `arg:"" help:"tenant ID to search"`
+}
+
+func (cmd *queryBlocksSummaryCmd) Run(ctx *globalOptions) error {
+	r, _, c, err := loadBackend(&cmd.backendOptions, ctx)
+	if err != nil {
+		return err
+	}
+
+	id, err := util.HexStringToTraceID(cmd.TraceID)
+	if err != nil {
+		return err
+	}
+
+	results, err := queryBucket(context.Background(), r, c, cmd.TenantID, id)
+	if err != nil {
+		return err
+	}
+
+	var (
+		combiner   = trace.NewCombiner()
+		// marshaller = new(jsonpb.Marshaler)
+		// jsonBytes  = bytes.Buffer{}
+	)
+
+	fmt.Println()
+	for i, result := range results {
+		combiner.ConsumeWithFinal(result.trace, i == len(results)-1)
+	}
+
+	combinedTrace, _ := combiner.Result()
+	size := 0
+	spanCount := 0
+	firstStartTime := combinedTrace.Batches[0].ScopeSpans[0].Spans[0].StartTimeUnixNano
+	lastEndTime := combinedTrace.Batches[0].ScopeSpans[0].Spans[0].EndTimeUnixNano
+	rootSpan := combinedTrace.Batches[0].ScopeSpans[0].Spans[0]
```

Comment:
the root span is not always the first. as you're iterating through all the spans below look for one with no parent span id

Reply:
instead of grabbing this first span like this we should init firstStartTime/lastEndTime/rootSpan/rootResource to better first values. something like this diff:

```
@@ -4,11 +4,14 @@ import (
        "bytes"
        "context"
        "fmt"
+       "math"
        "sort"
 
        "github.com/gogo/protobuf/jsonpb"
 
        "github.com/grafana/tempo/pkg/model/trace"
+       v1resource "github.com/grafana/tempo/pkg/tempopb/resource/v1"
+       v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
        "github.com/grafana/tempo/pkg/util"
 )
 
@@ -47,15 +50,16 @@ func (cmd *queryBlocksSummaryCmd) Run(ctx *globalOptions) error {
        }
 
        combinedTrace, _ := combiner.Result()
+
+       var rootSpan *v1.Span
+       var rootSpanResource *v1resource.Resource
        size := 0
        spanCount := 0
-       firstStartTime := combinedTrace.Batches[0].ScopeSpans[0].Spans[0].StartTimeUnixNano
-       lastEndTime := combinedTrace.Batches[0].ScopeSpans[0].Spans[0].EndTimeUnixNano
-       rootSpan := combinedTrace.Batches[0].ScopeSpans[0].Spans[0]
-       rootSpanResource := combinedTrace.Batches[0].Resource
+       firstStartTime := uint64(math.MaxUint64)
+       lastEndTime := uint64(0)
+
        rootServiceName := ""
        serviceNameMap := make(map[string]int)
-
        for _, b := range combinedTrace.Batches {
                size += b.Size()
                for _, attr := range b.Resource.Attributes {
```